### PR TITLE
feat: [PLG-1367]: Use new Stripe pricing when creating a subscription

### PR DESCRIPTION
**Pull request by thomas-grasman**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35709)

### CODE REVIEW

**Comment by thomas-grasman** [comment]
trigger messagecheck

File: 935-subscription-sdk/src/main/java/io/harness/subscription/helpers/impl/StripeHelperImpl.java [hunk 1]
```diff
@@ -157,6 +157,34 @@ public PriceCollectionDTO getPrices(ModuleType moduleType) {
     return toPriceCollectionDTO(priceResults);
   }
 
+  @Override
+  public Price getPrice(ModuleType moduleType, String type, String edition, String paymentFrequency, int quantity) {
+    String searchString = String.format(
```
Comment (pranil-harness) [inline 935-subscription-sdk/src/main/java/io/harness/subscription/helpers/impl/StripeHelperImpl.java:162]: Make this part of a constant?

File: 935-subscription-sdk/src/main/java/io/harness/subscription/helpers/impl/StripeHelperImpl.java [hunk 2]
```diff
@@ -157,6 +157,34 @@ public PriceCollectionDTO getPrices(ModuleType moduleType) {
...
+        PriceSearchParams.builder().setQuery(searchString).addAllExpand(Lists.newArrayList("data.tiers")).build();
+
+    List<Price> priceResults = stripeHandler.searchPrices(params).getData();
+
+    return priceResults.stream().findFirst().get();
+  }
+
+  @Override
+  public Price getPrice(ModuleType moduleType, String type, String edition, String paymentFrequency) {
+    String searchString = String.format(
```
Comment (pranil-harness) [inline 935-subscription-sdk/src/main/java/io/harness/subscription/helpers/impl/StripeHelperImpl.java:176]: Same here, add these templates as static members in a constant files, if we have one

File: 935-subscription-sdk/src/main/java/io/harness/subscription/helpers/impl/StripeHelperImpl.java (hunk 1 above)
Comment (thomas-grasman) [reply 935-subscription-sdk/src/main/java/io/harness/subscription/helpers/impl/StripeHelperImpl.java:162]: I don't understand. Do you want me to make a constant string for every possible search combination?

**Review by pranil-harness — COMMENTED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35709)* status: 3 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained.

**Review by pranil-harness — APPROVED** [review]

**Review by hekimoglu010 — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35709)* status: 1. unresolved discussions.

**Review by pranil-harness — APPROVED** [review]

**Review by thomas-grasman — COMMENTED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35709)* status: 2. unresolved discussions. 

___
*[`935-subscription-sdk/src/main/java/io/harness/subscription/helpers/impl/StripeHelperImpl.java` line 176 at r1](https://reviewable.io/reviews/harness/harness-core/35709#-N7rCPcvvobX20jhYzRS-r1-176:-N7vinc55ePZm1W6LTMR:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d01f3a3e195a9c72f6f53c7c9b8b163858869bfb/935-subscription-sdk/src/main/java/io/harness/subscription/helpers/impl/StripeHelperImpl.java#L176)):*
<details><summary><i>Previously, pranil-harness (Pranil Dasika) wrote…</i></summary><blockquote>

Same here, add these templates as static members in a constant files, if we have one
</blockquote></details>

Done.

**Review by hekimoglu010 — APPROVED** [review]

**Review by hekimoglu010 — APPROVED** [review]

**Review by devinder-harness — APPROVED** [review]

**Review by thomas-grasman — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35709#-:-N80jjHo8dj1SRe3jZs1:bnfp4nl)* status: 2. unresolved discussions.

**Review by thomas-grasman — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35709)* status: 2. unresolved discussions. 

___
*[`935-subscription-sdk/src/main/java/io/harness/subscription/helpers/impl/StripeHelperImpl.java` line 162 at r1](https://reviewable.io/reviews/harness/harness-core/35709#-N7rCPcvvobX20jhYzRS-r1-162:-N80juMI7L_rrnmhEwfa:bnfp4nl) ([raw file](https://github.com/harness/harness-core/blob/d01f3a3e195a9c72f6f53c7c9b8b163858869bfb/935-subscription-sdk/src/main/java/io/harness/subscription/helpers/impl/StripeHelperImpl.java#L162)):*
<details><summary><i>Previously, pranil-harness (Pranil Dasika) wrote…</i></summary><blockquote>

Make this part of a constant?
</blockquote></details>

:lgtm:
___
*[`935-subscription-sdk/src/main/java/io/harness/subscription/helpers/impl/StripeHelperImpl.java` line 176 at r1](https://reviewable.io/reviews/harness/harness-core/35709#-N7rCPcvvobX20jhYzRS-r1-176:-N80k0Sv7hHb-k-o5SOo:bnfp4nl) ([raw file](https://github.com/harness/harness-core/blob/d01f3a3e195a9c72f6f53c7c9b8b163858869bfb/935-subscription-sdk/src/main/java/io/harness/subscription/helpers/impl/StripeHelperImpl.java#L176)):*
<details><summary><i>Previously, thomas-grasman (Thomas Grasman) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

:lgtm:

**Pull request merged**